Add Push Notifications Capability

hannesmcman opened the pull request:
This PR is aimed at configuring the app to allow us to send remote push notifications to users using OneSignal. It may also include configuring local notifications using `react-native-push-notifications` if we want it to.

hannesmcman committed:
Configured OneSignal to send push notifications

hannesmcman committed:
prettier and flow

hannesmcman committed:
Merge branch 'master' into push-notifications

hawkrives commented on android/app/build.gradle:
```diff
@@ -219,6 +221,7 @@ if (propFile.canRead()) {
 }
 
 dependencies {
+    compile project(':react-native-onesignal')
```
Sort this

hawkrives commented on android/app/src/main/java/com/allaboutolaf/MainApplication.java:
```diff
@@ -11,6 +11,7 @@
 import com.BV.LinearGradient.LinearGradientPackage;
 import com.calendarevents.CalendarEventsPackage;
 import com.facebook.react.ReactApplication;
+import com.geektime.rnonesignalandroid.ReactNativeOneSignalPackage;
```
Sort this

hawkrives commented on android/app/src/main/java/com/allaboutolaf/MainApplication.java:
```diff
@@ -41,6 +42,7 @@ public boolean getUseDeveloperSupport() {
     protected List<ReactPackage> getPackages() {
       return Arrays.<ReactPackage>asList(
         new MainReactPackage(),
+            new ReactNativeOneSignalPackage(),
```
Sort and reindent this

hawkrives commented on android/settings.gradle:
```diff
@@ -1,4 +1,6 @@
 rootProject.name = 'AllAboutOlaf'
+include ':react-native-onesignal'
+project(':react-native-onesignal').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-onesignal/android')
```
Sort these

hawkrives commented on ios/AllAboutOlaf.xcodeproj/project.pbxproj:
```diff
@@ -1631,6 +1689,7 @@
 				LIBRARY_SEARCH_PATHS = (
 					"$(inherited)",
 					"\"$(SRCROOT)/$(TARGET_NAME)\"",
+					"\"$(SRCROOT)/$(TARGET_NAME)\"",
```
Remove dup

hawkrives commented on ios/AllAboutOlaf.xcodeproj/project.pbxproj:
```diff
@@ -1655,6 +1714,7 @@
 				LIBRARY_SEARCH_PATHS = (
 					"$(inherited)",
 					"\"$(SRCROOT)/$(TARGET_NAME)\"",
+					"\"$(SRCROOT)/$(TARGET_NAME)\"",
```
Remove dup

hawkrives commented on ios/AllAboutOlaf/AppDelegate.m:
```diff
@@ -52,6 +54,9 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 
+  self.oneSignal = [[RCTOneSignal alloc] initWithLaunchOptions:launchOptions
+                                                    appId:@"a46c6f2f-a240-4908-a359-801911e9b9ea"];
+
```
Remove extra line

hawkrives commented on ios/AllAboutOlaf/Info.plist:
```diff
@@ -87,9 +87,9 @@
+	<string>We only use your location when you would expect us to, and never when you aren't using the app.</string>
 	<key>NSLocationWhenInUseUsageDescription</key>
-	<string>We only use your location when you would expect us to, and never when you aren&apos;t using the app.</string>
+	<string>We only use your location when you would expect us to, and never when you aren't using the app.</string>
```
Undo

hawkrives commented on ios/AllAboutOlaf/Info.plist:
```diff
@@ -87,9 +87,9 @@
 	<string>We use your calendar to add events to your calendar so that you remember what you wanted to attend.</string>
 	<key>NSLocationAlwaysUsageDescription</key>
-	<string>We only use your location when you would expect us to, and never when you aren&apos;t using the app.</string>
+	<string>We only use your location when you would expect us to, and never when you aren't using the app.</string>
```
Undo

hawkrives reviewed (dismissed):
A bunch of little tweaks to the native project files

hannesmcman committed:
Minor tweaks to native project files

hawkrives commented on android/app/build.gradle:
```diff
@@ -229,6 +231,7 @@ dependencies {
     implementation project(':react-native-keychain')
     implementation project(':react-native-linear-gradient')
     implementation project(':react-native-network-info')
+    compile project(':react-native-onesignal')
```
Change this to `implementation`

hawkrives commented on ios/AllAboutOlaf/AppDelegate.m:
```diff
@@ -52,6 +54,8 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   [[AVAudioSession sharedInstance] setCategory: AVAudioSessionCategoryPlayback error: nil];
 
+  self.oneSignal = [[RCTOneSignal alloc] initWithLaunchOptions:launchOptions
+                                                    appId:@"a46c6f2f-a240-4908-a359-801911e9b9ea"];
```
Align the `:` here, please.

```objc
self.oneSignal = [[RCTOneSignal alloc] initWithLaunchOptions:launchOptions
                                                       appId:@"a46c6f2f-a240-4908-a359-801911e9b9ea"];
```

hawkrives reviewed (changes requested):
I don't think I have any more nitpicky comments after this.

hannesmcman committed:
done with the nitpicky?

hannesmcman committed:
Added OneSignal Extension (build should fail)

rye committed:
Merge branch 'master' into push-notifications

Conflicts:
	Gemfile.lock
	- Resolved by taking latest Fastlane
	source/app.js
	- Kept branch-local changes
	yarn.lock
	- Reran yarn

drewvolz commented on source/app.js:
```diff
@@ -30,6 +31,35 @@ function getCurrentRouteName(navigationState: NavigationState): ?string {
 
 export default class App extends React.Component<Props> {
+	componentDidMount() {
+		startStatusBarColorChanger()
```
Where is the start/stop coming from? Currently it breaks things.

drewvolz replied:
I feel like it might be related to the way we were telling the iOS status bar to be dark or light? @hawkrives

hawkrives committed:
Merge branch 'master' into push-notifications

* master: (116 commits)
  Fix the somewhat normal "❤️" notice
  Add some... interesting Home screen notices
  use "an approximation of the default behavior for iOS" for header transitions
  extend PureComponent in StyledAlphabetListView
  debounce searching in SearchableAlphabetListView by 200ms instead of 50
  use React.PureComponent in CourseResultsList
  Remove font boldening from selectable text
  Release v2.6.0, beta 5
  Disable autocorrect in controlled searchbar
  prettify
  remove mem from deps
  switch from "mem" to lodash/memoize
  extract doSearch from CourseResultsList
  collapse memoizedDoSearch
  stop lowercasing query twice
  chore(package): update lockfile
  chore(package): update danger to version 3.8.4
  Update snapshot tests
  Replace outline with empty favorites icon
  update Course's recents list to use c.iosPushButtonCellForeground color
  ...

hawkrives committed:
fix merge commit conflict problems from 696933922be7247ea998da2c4c87dcfabca16e1c

hawkrives committed:
try enabling automatic code signing

hawkrives committed:
Merge branch 'master' into push-notifications

hawkrives committed:
disable automatic signing again

hawkrives committed:
update push extension certs

hawkrives committed:
add apple_dev_portal_id field to Appfile for logging in to the dev portal

hawkrives committed:
add new lane to fetch certs for both the app and push extension

hawkrives committed:
add CODE_SIGN_ENTITLEMENTS to xcodeproj

hawkrives committed:
add new onesignal / app creation steps to fastlane

hawkrives committed:
make generate_pem public

hawkrives committed:
update push extension bundle ID to fix xcode error

❌  error: Embedded binary's bundle identifier is not prefixed with the
parent app's bundle identifier.

hawkrives committed:
fix bundle-data script when building from xcode

hawkrives committed:
fix looking for the wrong extension id when running match

hawkrives reviewed (approved):
sure
